pkg/metrics/node.go

```diff
@@ -0,0 +1,182 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	mgmtcontrollers "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"github.com/rancher/wrangler/pkg/ticker"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	nodeProviderLabel  = "provider"
+	nodeClusterIDLabel = "cluster_id"
+	nodeIsActiveLabel  = "is_active"
+)
+
+var (
+	nodeLabels = []string{nodeProviderLabel, nodeClusterIDLabel, nodeIsActiveLabel}
+	numNodes   = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: "cluster_manager",
+			Name:      "nodes",
+			Help:      "Number of nodes in rancher managed clusters",
+		}, nodeLabels,
+	)
+	numCores = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: "cluster_manager",
+			Name:      "node_cores",
+			Help:      "Number of node cores in rancher managed clusters",
+		}, nodeLabels,
+	)
+)
+
+const (
+	reportInterval = time.Duration(15 * time.Second)
+	logPrefix      = "[prometheus-node-metrics]"
+)
+
+type nodeMetrics struct {
+	nodeCache    mgmtcontrollers.NodeCache
+	clusterCache mgmtcontrollers.ClusterCache
+}
+
+func (m *nodeMetrics) collect(ctx context.Context) {
+	for range ticker.Context(ctx, reportInterval) {
+		logrus.Debugf("%s collecting nodes to report metrics", logPrefix)
+
+		nodes, err := m.nodeCache.List("", labels.NewSelector())
```

Comment:
Question here: does labels.NewSelector() return a selector that matches everything (so same as labels.Everything())? That would be my guess but I'd like to be sure.

Reply:
Yea, but `labels.Everything()` conveys better intent, I'll update